src/core/qgssqlstatement.cpp

```diff
@@ -89,8 +91,8 @@ QString QgsSQLStatement::quotedIdentifierIfNeeded( const QString &name )
       return quotedIdentifier( name );
     }
   }
-  static const QRegExp IDENTIFIER_RE( "^[A-Za-z_\x80-\xff][A-Za-z0-9_\x80-\xff]*$" );
-  return IDENTIFIER_RE.exactMatch( name ) ? name : quotedIdentifier( name );
+  static const QRegularExpression IDENTIFIER_RE( "^[A-Za-z_\x80-\xff][A-Za-z0-9_\x80-\xff]*$" );
```

Comment:
maybe ?
```suggestion
  const thread_local QRegularExpression IDENTIFIER_RE( "^[A-Za-z_\x80-\xff][A-Za-z0-9_\x80-\xff]*$" );
```
(if we have no guarantee that QRegularExpression::match() can be safely called from multiple threads on the same object), or remove the ``static`` qualitifer

Reply:
Yeah, you're right, I've used thread_local elsewhere.